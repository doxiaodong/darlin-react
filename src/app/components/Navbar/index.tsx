import * as React from 'react'
import { Link } from 'react-router'

export class CompNavbar extends React.Component<{}, {}> {

  render() {
    return (
      <div>
        <Link to='/'>首页</Link>
        <Link to='/article/all'>文章列表</Link>
      </div>
    )
  }
}
