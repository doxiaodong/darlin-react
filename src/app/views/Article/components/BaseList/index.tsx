import * as React from 'react'
// import { Link } from 'react-router'
import {
  List,
  ListItem
} from 'material-ui/List'
import Subheader from 'material-ui/Subheader'
import { transformDate } from 'app/base/transform/date'

import * as style from './style.scss'

export class CompBaseList extends React.Component<{ articles: any[] }, {}> {

  render() {
    const listItem = this.props.articles.map((article) =>
      (
        <ListItem key={article.url}>
          <div className={style.article}>
            <div className={style.title}>{article.title}</div>
            <div>{transformDate(article.createTime)}</div>
          </div>
        </ListItem>
      )
    )
    return (
      <List>
        <Subheader>文章列表</Subheader>
        {listItem}
      </List>
    )
  }
}
