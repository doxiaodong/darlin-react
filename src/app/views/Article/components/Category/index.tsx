import * as React from 'react'
// import { Link } from 'react-router'
import {
  List,
  ListItem
} from 'material-ui/List'
import Subheader from 'material-ui/Subheader'
import { observer } from 'mobx-react'

import { CategoryStore } from './store'

@observer
export class CompCategory extends React.Component<{ categoryStore: CategoryStore }, {}> {

  componentDidMount() {
    this.props.categoryStore.getArticleCategories()
  }

  render() {
    const listItem = this.props.categoryStore.categories.map((category) => {
      return (
        <ListItem key={category.key}>
          {category.name}
        </ListItem>
      )
    })
    return (
      <List>
        <Subheader>分类</Subheader>
        {listItem}
      </List>
    )
  }
}
