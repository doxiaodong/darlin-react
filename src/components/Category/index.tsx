import * as React from 'react'
import {
  withRouter,
  InjectedRouter
} from 'react-router'
import {
  List,
  ListItem,
  makeSelectable
} from 'material-ui/List'
import Subheader from 'material-ui/Subheader'
import { observer } from 'mobx-react'
import categoryStore from 'stores/category'
import articleStore from 'stores/article'

import * as style from './style.scss'

const SelectableList = makeSelectable(List)

@observer
export class CompCategory extends React.Component<{
  params?: any,
  router?: InjectedRouter
}, {}> {

  handleRequestChange = (event, key) => {
    categoryStore.handleRequestChange(event, key)
    this.props.router.push({
      pathname: `/article/${key}`
    })
    articleStore.getArticleList(key)
  }

  componentDidMount() {
    categoryStore.getArticleCategories()
    categoryStore.selectedCategoryKey = this.props.params.category
  }

  render() {
    const listItem = categoryStore.categories.map((category) => {
      return (
        <ListItem key={category.key} value={category.key}>
          <div className={category.key === categoryStore.selectedCategoryKey ? style.active : ''}>
            {category.name}
          </div>
        </ListItem>
      )
    })
    return (
      <div className={style.container}>
        <div className={style.right}>
          <SelectableList
            value={categoryStore.selectedCategoryKey}
            onChange={this.handleRequestChange}
          >
            <Subheader>分类</Subheader>
            {listItem}
          </SelectableList>
        </div>
        <div className={style.left} />
      </div>
    )
  }
}

export const CompCategoryWithRouter = withRouter(CompCategory)
