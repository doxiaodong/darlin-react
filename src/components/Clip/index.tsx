import * as React from 'react'
import * as Clipboard from 'clipboard'

export class Clip extends React.Component<{ value, onClip }, void> {

  content: string = ''
  clipboard: Clipboard

  componentWillUnmount() {
    if (this.clipboard) {
      this.clipboard.destroy()
    }
  }

  ref = (element) => {
    if (!element) {
      return
    }
    const { onClip } = this.props

    const option: Clipboard.Options = {
      text: (ele) => {
        return this.content
      }
    }
    this.clipboard = new Clipboard(element, option)

    this.clipboard.on('success', () => {
      onClip(true)
    })
      .on('error', () => {
        onClip(false)
      })
  }

  render() {
    this.content = this.props.value
    return (
      <div ref={this.ref}>
        {this.props.children}
      </div>
    )
  }
}
