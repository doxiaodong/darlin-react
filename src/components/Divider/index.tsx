import * as React from 'react'

import * as style from './style.scss'

export class Divider extends React.Component<{}, {}> {
  render() {
    return (
      <hr className={style.hr} />
    )
  }
}
