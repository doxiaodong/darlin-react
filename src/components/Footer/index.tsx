import * as React from 'react'
import { observer } from 'mobx-react'
import i18nStore from 'stores/i18n'
import store from './store'

import * as style from './style.scss'

@observer
export class Footer extends React.Component<void, void> {

  componentDidMount() {
    document.addEventListener('click', store.close)
  }

  componentWillUnmount() {
    document.removeEventListener('click', store.close)
  }

  stopPropagation = (event) => {
    event.nativeEvent.stopImmediatePropagation()
  }

  render() {
    const { t } = i18nStore
    const listItem = store.langs.map((lang) => {
      return (
        <li
          key={lang.key}
          className={`${style.li} a` + (store.selectedKey === lang.key ? ` ${style.active}` : '')}
          onClick={() => { store.changeLanguage(lang.key) }}
        >
          {lang.word}
        </li>
      )
    })
    const list = <ul className={style.ul}>{listItem}</ul>
    return (
      <footer className={style.footer}>
        <div className={style.content}>
          <a href="http://www.miitbeian.gov.cn/" target="_blank">蜀ICP备15013600号-1</a> @<a>react.darlin.me</a>
          <div className={style.i18n} onClick={this.stopPropagation}>
            <p className="a" onClick={store.toggleShow}>{t('common:langSetting')}</p>
            {store.showLang && list}
          </div>
        </div>
      </footer>
    )
  }
}
