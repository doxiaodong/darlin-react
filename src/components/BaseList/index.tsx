import * as React from 'react'
// import { Link } from 'react-router'
import { observer } from 'mobx-react'
import {
  List,
  ListItem
} from 'material-ui/List'
import RaisedButton from 'material-ui/RaisedButton'
import { Subheader } from 'components/Subheader'
import { Divider } from 'components/Divider'
import { transformDate } from 'base/transform/date'
import articleStore from 'stores/article'
import i18nStore from 'stores/i18n'

import * as style from './style.scss'

@i18nStore.namespace(['article'])
@observer
export class BaseList extends React.Component<{ articles: any[] }, {}> {

  render() {
    const { articles } = this.props
    const { t } = i18nStore
    const listItem = articles.map((article, index) =>
      (
        <div key={article.url}>
          {index > 0 && <Divider />}
          <ListItem>
            <div className={style.article}>
              <div className={style.title}>{article.title}</div>
              <div>{transformDate(article.createTime)}</div>
            </div>
          </ListItem>
        </div>
      )
    )
    const moreButton = (
      <div className={style.more}>
        <RaisedButton
          label={t('article:loadMore')}
          primary={true}
          onClick={articleStore.getMoreArticles}
        />
      </div>
    )
    return (
      <div className='each-block'>
        <Subheader title={t('article:articleList')} />
        <List style={{ paddingTop: 0 }}>
          {listItem}
          {(articleStore.hasMore || articles.length === 0) && <Divider />}
          {articles.length === 0 && <ListItem>{t('article:noArticle')}</ListItem>}
          {articleStore.hasMore && moreButton}
        </List>
      </div>
    )
  }
}
