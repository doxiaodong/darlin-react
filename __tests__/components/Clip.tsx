import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { Clip } from 'components/Clip'

describe('App Component', () => {
  test('Snapshot', () => {
    const component = renderer.create(
      <Clip value={1} onClip={() => null}>copy</Clip>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('Renders children', () => {
    const onClip = jest.fn()
    const wrapper = shallow(
      <Clip value={1} onClip={onClip}>
        <div className="copy">copy</div>
      </Clip>
    )
    expect(wrapper.find('.copy')).toHaveLength(1)

    // what a fuck
    wrapper.find('.copy').simulate('click')
    onClip(false)

    expect(onClip).toHaveBeenCalled()
  })
})
