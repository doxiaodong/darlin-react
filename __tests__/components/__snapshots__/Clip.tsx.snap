// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot 1`] = `
<div>
  copy
</div>
`;
